pebble/pathapp/src/pathapp.c

```diff
@@ -0,0 +1,285 @@
+#include "pebble.h"
+
+#define MAXBUFFERSIZE 200
+#define MAX_SCHED_ITEMS 10
+#define MAX_STN_ITEMS 20
+#define MAX_DIR_ITEMS 5
+
+#define STN_NAME_LEN 10
+
+static Window *main_window, *direction_window, *schedule_window;
+static SimpleMenuLayer *stn_layer, *dir_layer, *sched_layer;
+
+static SimpleMenuItem stn_items[MAX_STN_ITEMS], dir_items[MAX_DIR_ITEMS], sched_items[MAX_SCHED_ITEMS];
+static SimpleMenuSection stn_sections[1], dir_sections[1], sched_sections[1];
+
+static char buffer[MAXBUFFERSIZE];
+
+static char from_station[STN_NAME_LEN];
+static char *delim = ";";
+
+enum {
+	PATH_STN_KEY = 0x0,
+};
+
+static size_t sched_num_items;
+
+static char *stn_names[] = {
+	"JSQ",
+	"Grove St",
+	"Newport",
+	"Hoboken",
+	"Chris St",
+	"9 St",
+	"14 St",
+	"23 St",
+	"33 St",
+	NULL,
+};
+
+struct direction {
+	char *title;
+	char *stub;
+};
+
+static struct direction directions[] = {
+	{ .title = "JSQ to 33rd", .stub = "jsq_33rd" },
```

Comment:
We discussed this offline. It would less confusing to just see destination instead of source-destination, since the user of the app needs to know only 2 things: where he is at and where he wants to go.